### PR TITLE
Feature/scheduled backups

### PR DESCRIPTION
**Pull request by kusalananda**
Add a shell script for scheduled (cron) backup of database data, logs, and uploaded data.

The script is `scripts/scheduled-backup.sh` and it performs an optional database dump using `scripts/database-backup.sh` (which was renamed and enhanced slightly to make it possible to choose a backup format that is compressed).

```
Usage:

    ./scripts/scheduled-backup.sh -h

    ./scripts/scheduled-backup.sh [-n] [-v]

Options:

    -h  displays this help text
    -n  do not dump database
    -v  be more verbose
```

The script creates incremental backups under `backups/`:

```
$ ls -F backups
backup-20210507-175737/       backup-20210508-123104/       backup-20210508-182147/
backup-20210507-180837/       backup-20210508-123159/       backup-20210508-182201/
backup-20210507-180839/       backup-20210508-135416/       backup-20210508-182337/
backup-20210507-181006/       backup-20210508-135430/       backup-20210508-182650/
backup-20210507-181513/       backup-20210508-135536/       backup-20210508-183543/
backup-20210507-181602/       backup-20210508-135544/       backup-20210508-183605/
backup-20210507-192941/       backup-20210508-181813/       latest@
backup-20210508-121035/       backup-20210508-182135/
```
```
$ ls -F backups/latest/
db/      uploads/
```

Part of the backup process uses `rsync`, and also `rsync` in the container, so I updated the Docker compose files to install `rsync`.

### CODE REVIEW

**Comment by kusalananda** [comment]
This is a draft pull request. We still need to make sure that logs are properly backed up.

File: scripts/scheduled-backup.sh [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+fi >&2
+
+if ! command -v rsync >/dev/null 2>&1
+then
+	echo 'rsync is not available, or not in PATH' >&2
+	exit 1
+fi >&2
+
+unset topdir
+topdir=$( readlink -f "${topdir:-"$( dirname "$0" )/.."}" )
```
Comment (pontus) [inline scripts/scheduled-backup.sh:108]: Is `topdir` magical in some sh-incarnation? The use after unset confuses me.

**Review by pontus — APPROVED** [review]
:+1:

File: scripts/scheduled-backup.sh (hunk 1 above)
Comment (kusalananda) [reply scripts/scheduled-backup.sh:108]: No it's not.  I will make that code less confusing right away. Thanks.

Comment (kusalananda) [reply scripts/scheduled-backup.sh]: Sorted out in a3d35b8

**Comment by pragermh** [comment]
Thanks @kusalananda. When I run kusalananda:feature/scheduled-backups with `docker-compose build --no-cache && docker-compose up ` a 'db' dir is created, although it has been renamed to 'db-backup', and i get db-related error msgs, e.g: `asv-db     | 2021-05-17 08:56:53.550 UTC [38] ERROR: role "anon" does not exist`.

**Comment by kusalananda** [comment]
> Thanks @kusalananda. When I run kusalananda:feature/scheduled-backups with `docker-compose build --no-cache && docker-compose up ` a 'db' dir is created, although it has been renamed to 'db-backup', and i get db-related error msgs, e.g: `asv-db | 2021-05-17 08:56:53.550 UTC [38] ERROR: role "anon" does not exist`.

The `db` directory seems to be created when issueing the command `docker-compose up`.  It seems as if this is a bind-mounted directory that the container expects to find certain files in.  These files are now located in `db-backup`.  Since they don't have anything to do with database backups (and never had), I'm moving them back into `db`.

**Pull request merged**